components/sync_manager/src/syncmanager.udl

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace syncmanager { };
+
+[Error]
+enum SyncManagerError {
+    "UnknownEngine",
+    "UnsupportedFeature",
+    "Sync15Error",
+    "UrlParseError",
+    "InterruptedError",
+    "JsonError",
+    "LoginsError",
+    "PlacesError",
+    "AnyhowError",
+};
+
+dictionary SyncParams {
+    // Why are we performing this sync?
+    SyncReason reason;
+    // Which engines should we sync?
+    SyncEngineSelection engines;
+    // TODO
+    record<DOMString, boolean> enabled_changes;
+    // Keys to encrypt/decrypt data from local database files.  These are
+    // separate from the key we use to encrypt the sync payload as a whole.
+    record<DOMString, string> local_encryption_keys;
+    // Authorization for the sync server
+    SyncAuthInfo auth_info;
+    // TODO
+    string? persisted_state;
+    // TODO
+    DeviceSettings device_settings;
```

Comment:
Information about the current device, such as its name, formfactor and FxA device ID.